addon/ng2/commands/newroute.ts

```diff
@@ -0,0 +1,85 @@
+'use strict';
+
+import * as path from 'path';
+import * as fs from 'fs';
+import * as Promise from 'ember-cli/lib/ext/promise';
+import * as Blueprint from 'ember-cli/lib/models/blueprint';
+import * as EmberGenerateCommand from 'ember-cli/lib/commands/generate';
+import * as SilentError from 'silent-error';
+import * as _ from 'lodash';
+
+var EOL = require('os').EOL;
+const chalk = require('chalk');
+
+const newRouteCommand = EmberGenerateCommand.extend({
+    name: 'newroute',
+    description: 'generates a new router',
+    works: 'insideProject',
+    availableOptions : [ ],
+
+    beforeRun: function(rawArgs) {
+      if (!rawArgs.length) {
+        return;
+      }
+
+    EmberGenerateCommand.prototype.printDetailedHelp = function (options) {
+      var blueprintList = fs.readdirSync(path.join(__dirname, '..', 'blueprints'));
+      var blueprints = blueprintList
+        .filter(bp => bp.indexOf('routes') !== -1)
+        .map(bp => Blueprint.load(path.join(__dirname, '..', 'blueprints', bp)));
+      
+      var output = '';
+      blueprints
+        .forEach(function (bp) {
+          output += bp.printBasicHelp(false) + EOL;
+        });
+      this.ui.writeLine(chalk.cyan('  Available blueprints'));
+      this.ui.writeLine(output);
+    };
+    return EmberGenerateCommand.prototype.beforeRun.apply(this, arguments);
+  },
+
+  run: function(commandOptions, rawArgs){
+    var blueprintName = rawArgs[0];
+
+    if (!blueprintName) {
+      return Promise.reject(new SilentError('The `ember generate` command requires a ' +
+                                            'blueprint name to be specified. ' +
+                                            'For more details, use `ember help`.'));
+    }
+    if (blueprintName !== 'routes'){
+      return Promise.reject(new SilentError(`Unknown blueprint name ${blueprintName} for newroute. ` +
+                                            `Try the routes blueprint`))
+    }
+
+    var Task = this.tasks.GenerateFromBlueprint;
+    var task = new Task({
+      ui: this.ui,
+      analytics: this.analytics,
+      project: this.project,
+      testing: this.testing,
+      settings: this.settings
+    });
+
+    var taskArgs = {
+      args: rawArgs
+    };
+
+    if (this.settings && this.settings.usePods && !commandOptions.classic) {
+      commandOptions.pod = !commandOptions.pod;
+    }
+
+    var taskOptions = _.merge(taskArgs, commandOptions || {});
+
+    if (this.project.initializeAddons) {
```

Comment:
What does this step do? Can you look at other Angular CLI commands to see if it's needed?